File: add/codesnippet/CPP/e-system.configuration.a_1_1.cpp
    private:
        void FormSettings_SettingsSaving(Object^ sender,
            CancelEventArgs^ e)
        {
            //Should check for settings changes first.
            ::DialogResult^ dialogResult = MessageBox::Show(
                "Save current values for application settings?",
                "Save Settings", MessageBoxButtons::YesNo);
            if (::DialogResult::No == dialogResult)
            {
                e->Cancel = true;
            }
        }

File: add/codesnippet/CSharp/e-system.configuration.a_1_1.cs
        void frmSettings1_SettingsSaving(object sender, CancelEventArgs e)
        {
            //Should check for settings changes first.
            DialogResult dr = MessageBox.Show(
                            "Save current values for application settings?",
                            "Save Settings", MessageBoxButtons.YesNo);
            if (DialogResult.No == dr)
            {
                e.Cancel = true;
            }
        }